sklearnex/preview/linear_model/logistic_regression.py

```diff
@@ -0,0 +1,329 @@
+            onedal_params = {
+                "tol": self.tol,
+                "C": self.C,
+                "fit_intercept": self.fit_intercept,
+                "solver": self.solver,
+                "max_iter": self.max_iter,
+            }
+            self._onedal_estimator = onedal_LogisticRegression(**onedal_params)
+
+        def _onedal_fit(self, X, y, sample_weight, queue=None):
+            assert sample_weight is None
+
+            check_params = {
+                "X": X,
+                "y": y,
+                "dtype": [np.float64, np.float32],
+                "accept_sparse": False,
+                "multi_output": False,
+                "force_all_finite": True,
+            }
+            if sklearn_check_version("1.2"):
+                X, y = self._validate_data(**check_params)
+            else:
+                X, y = check_X_y(**check_params)
+            self._initialize_onedal_estimator()
+            try:
+                self._onedal_estimator.fit(X, y, queue=queue)
+                self._save_attributes()
+            except RuntimeError:
+                logging.getLogger("sklearnex").info(
+                    f"{self.__class__.__name__}.fit "
+                    + get_patch_message("sklearn_after_onedal")
+                )
+
+                del self._onedal_estimator
+                super().fit(X, y)
+
+        def _onedal_predict(self, X, queue=None):
+            X = self._validate_data(X, accept_sparse=False, reset=False)
+            if not hasattr(self, "_onedal_estimator"):
+                self._initialize_onedal_estimator()
+                self._onedal_estimator.coef_ = self.coef_
+                self._onedal_estimator.intercept_ = self.intercept_
+                self._onedal_estimator.classes_ = self.classes_
+
+            return self._onedal_estimator.predict(X, queue=queue)
+
+        def _onedal_predict_proba(self, X, queue=None):
+            X = self._validate_data(X, accept_sparse=False, reset=False)
+            if not hasattr(self, "_onedal_estimator"):
+                self._initialize_onedal_estimator()
+                self._onedal_estimator.coef_ = self.coef_
+                self._onedal_estimator.intercept_ = self.intercept_
+
+            return self._onedal_estimator.predict_proba(X, queue=queue)
+
+        def _onedal_predict_log_proba(self, X, queue=None):
+            X = self._validate_data(X, accept_sparse=False, reset=False)
+            if not hasattr(self, "_onedal_estimator"):
+                self._initialize_onedal_estimator()
+                self._onedal_estimator.coef_ = self.coef_
+                self._onedal_estimator.intercept_ = self.intercept_
+
+            return self._onedal_estimator.predict_log_proba(X, queue=queue)
+
+else:
+    from daal4py.sklearn.linear_model import LogisticRegression
+
+    logging.warning(
+        "Sklearnex LogisticRegression requires oneDAL version >= 2024.0.1 "
+        "but it was not found"
+    )
```

Comment:
We require this because adding daal_check_version to dispatcher is not a pattern for our repository https://github.com/intel/scikit-learn-intelex/blob/master/sklearnex/dispatcher.py#L65-L83 
You can see this pattern in other newly added algorithms:
https://github.com/intel/scikit-learn-intelex/blob/master/sklearnex/preview/cluster/k_means.py#L365-L370
https://github.com/intel/scikit-learn-intelex/blob/master/sklearnex/linear_model/linear.py#L367-L373
This helps to avoid public CI errors when linked with older oneDAL version.

Reply:
Removing this will also lead to adding lots of if-statements in tests that we don't have at the moment. That's why I'm not sure about this

Reply:
I think it can probably be included for now since it matches our other preview algos. If it should be removed then a separate task can be created for it.